scipy/spatial/tests/test_hausdorff.py

```diff
@@ -155,6 +156,13 @@ def test_massive_arr_overflow():
     # on 64-bit systems we should be able to
     # handle arrays that exceed the indexing
     # size of a 32-bit signed integer
+    try:
+        import psutil
+    except ModuleNotFoundError:
+        pytest.skip("psutil required to check available memory")
+    if psutil.virtual_memory().available < 80*2**30:
+        # Don't run the test if there is less than 80 gig of RAM available.
+        pytest.skip('insufficient memory available to run this test')
```

Comment:
FWIW, I wrote the test and developed it on a machine with 128 GB of memory. I should probably double check that this doesn't skip on that machine.

That said, this does feel awkward--I can't decide if it would be better long-term to have a high memory decorator/skipper like this, or to avoid adding these tests. At the time, having a regression test that almost never runs, but can be occasionally run to check for a regression, was appealing. On the flip side, this is a bit like generating tremendous amounts of data to verify that there isn't a typing error in the Cython code. I bet there are probably cases where we've fixed such errors without adding an expensive regression test, and I suppose if you were writing the code from scratch you'd probably opt *not* to include a high-memory test to look for such a typing error.

It also seems like something a static analyzer might be able to catch, but it would have to verify type propagation across the Cython language interface and into the types in the Cython function bodies I think.

Probably a bad idea, but some kind of separate test suite for expensive/functional/crazy but useful tests somewhere that you run once a year or something? I dunno..

Reply:
In the absence of a really intelligent cross-language type checker, it is also worth noting that the test should fail almost instantly before the fix is applied, so you probably could just allow it to pass if there is no typing failure raised within < 1 second (would need to double check pass/fail threshold I guess). Of course, that's also not fully satisfying..

Reply:
I don't have a preference among the various alternatives to this pull request (except that recent experience suggests that timeout-based unit tests are fragile).  Of course, simply removing the test would be the easiest :smiley:.  I don't think merging this PR precludes eventually implementing any of the alternatives, but even if folks approve of the change, it shouldn't be merged until @tylerjereddy has a chance to verify that the revised test in fact runs and passes on the system with 128GB.

Reply:
Regarding this test, I don't mind either way (this PR seems fine, removing the test too).

> lol does it even make sense to have the test there then? If no CI can run it. Just seems like some dead code for now.
>
> > Probably a bad idea, but some kind of separate test suite for expensive/functional/crazy but useful tests somewhere that you run once a year or something? I dunno..
> 
> Maybe not that bad of an idea. Our test suite is huge and we could for sure separate things.

These comments I disagree with. It is not dead code, you can run it locally - and multiple devs indeed do this. Having it marked `xslow` and in the main repo is much nicer for both usability and for maintenance effort than setting up a separate repo.